docs/guide/advanced/navigation-failures.md

```diff
@@ -0,0 +1,57 @@
+# Navigation Failures
+
+When using `router-link`, Vue Router internally calls `router.push` to trigger a navigation. Depending on the current location and existing [Navigation Guards](./navigation-guards.md), this navigation might end up in a new page being shown, but there are a couple of situations where we will stay on the same page:
+
+- We are already on the page we are trying to go to
+- A navigation guard aborts the navigation by calling `next(false)`
+- A navigation guard throws an error or calls `next(new Error())`
+
+When using a regular `router-link`, **none of these failures will log an error**. However, if you are using `router.push` or `router.replace`, you might come across an _"Uncaught (in promise) Error"_ message followed by a more specific message in your console. This is part of the _Navigation Failures_ system in Vue Router and it is only available from version 3.2.0 onwards but existed for a long time before: through the two optional callbacks, `onComplete` and `onAbort` that can be passed to `router.push`. Since version 3.1.0, `router.push` and `router.replace` return a _Promise_ if no `onComplete`/`onAbort` callback is provided. This _Promise_ resolves instead of invoking `onComplete` and rejects instead of invoking `onAbort`.
+
+## Detecting Navigation Failures
+
+_Navigation Failures_ are `Error` instances with a few extra properties. To check if an error comes from the Router, use the `isRouterError` function:
+
+```js
+import { NavigationFailureType, isRouterError } from 'vue-router'
+
+// trying to access the admin page
+router.push('/admin').catch(failure => {
+  if (failure) {
```

Comment:
Is there a case where this would be _falsy_ even though the `catch` was triggered?

Reply:
No, that shouldn't happen, so I think we can remove this!

Reply:
Great!